### PR TITLE
Podspec

### LINKED ISSUE #49: Podspecs
Hi can u make podspecs for this?

### PR DESCRIPTION
**Pull request by mcrumm**
Fixes #49 

I was getting a build error until I added the `AppDelegate.h` file, but its contents are straight out of a `react-native init` project, so I'm not sure that this is syntactically the best way to go when loading the ObjC code via CocoaPods.

If someone with more Xcode experience could chime in, here, I'd appreciate it.

### CODE REVIEW

**Comment by JAStanton** [comment]
20 days old.. lovely. Can you bump the version on it? @mcrumm :)

**Comment by mcrumm** [comment]
@JAStanton Done!  Thanks for taking a look at this.

File: RCTOrientation.podspec [hunk 1]
```diff
@@ -0,0 +1,12 @@
+Pod::Spec.new do |s|
+  s.name         = "RCTOrientation"
```
Comment (JAStanton) [inline RCTOrientation.podspec]: ```  s.name         = "react-native-orientation"```

File: RCTOrientation/AppDelegate.h [hunk 2]
```diff
@@ -0,0 +1,16 @@
+/**
```
Comment (JAStanton) [inline RCTOrientation/AppDelegate.h]: Is this file needed?

File: RCTOrientation.podspec (hunk 1 above)
Comment (mcrumm) [inline RCTOrientation.podspec]: If we change the name here, we'll also need to rename this file to `react-native-orientation.podspec`, so users can do this:

```
 pod 'react-native-orientation', :path => '../node_modules/react-native-orientation'
```

Instead of this:

```
 pod 'react-native-orientation', :podspec => '../node_modules/react-native-orientation/RCTOrientation.podspec'
```

Does that make sense?

File: RCTOrientation/AppDelegate.h (hunk 2 above)
Comment (mcrumm) [inline RCTOrientation/AppDelegate.h]: As I mentioned in the PR note, I was getting a build error without it. I'm an XCode/Cocoapods newb, though, so I'm not sure it's strictly necessary, or if there's something missing, configuration-wise, that would eliminate the need for duplicating this here.

Comment (mcrumm) [inline RCTOrientation/AppDelegate.h]: @JAStanton I believe this is necessary because of the AppDelegate import in [Orientation.m](https://github.com/yamill/react-native-orientation/blob/master/RCTOrientation/Orientation.m#L6), but that's all I've got. 😕

**Comment by nevir** [comment]
👎 Injecting behavior into `AppDelegate` via a category is problematic:

* Not all apps call their application delegate `AppDelegate` (at which point this package becomes unusable)
* Apps may need to override the behavior of `supportedInterfaceOrientationsForWindow`, but this package prevents it.
* Using this as a cocoapod would not work if the user has `use_frameworks!` enabled in their `Podspec` (since it's dynamically linked, rather than statically linked into the binary that defines `AppDelegate`)
* If someone changes their header search paths, there's a possiblitity that this embedded `AppDelegate` _will override the one defined in their app_, leading to compile failures if the headers declare different properties.

---

IMO a much cleaner and more friendly approach is to just expose `[RCTOrientation getOrientation]`, and direct consumers of this package to [implement that delegate method themselves](https://github.com/yamill/react-native-orientation/blob/master/RCTOrientation/Orientation.m#L11-L13).

**Comment by mcrumm** [comment]
@nevir Thanks for the explanation!  I assumed my newb-ness would cause problems here, but not of that magnitude 😄 

@yamill Would you like to weigh in, here?  I would be happy to attempt refactoring this to support CocoaPods, but only if you're interested in adding that support.

**Comment by JAStanton** [comment]
I'm in the process right now to get this to work: https://github.com/convoyinc/react-native-orientation

**Comment by JAStanton** [comment]
I'll see what I can do Michael! Thanks for getting the ball rolling! I'll
update this when I get something working

**Comment by nevir** [comment]
Feel free to steal commits back, imo :)

**Comment by mcrumm** [comment]
@nevir See #75 😃
